### DISCUSSION TITLE
How to remove(delete) highlights

### DISCUSSION THREAD

**Discussion by xiaolibuzai-ovo** [pymupdf/PyMuPDF#4111] [Looking for help]
[chatgptxx.pdf](https://github.com/user-attachments/files/18002871/chatgptxx.pdf)
<img width="1404" alt="image" src="https://github.com/user-attachments/assets/820323b8-f0f1-45f3-a3ef-c3184f7bce75">
For example, these borders.
thanks for you help！

**Comment by xiaolibuzai-ovo** [comment]
oh, maybe this is links

```python
    links=page.get_links()
    for link in links:
        page.delete_link(link)
```

**Comment by JorjMcKie** [comment]
You can use redactions for this.
1. For rectangles like your 2 large red one do: `page.add_redact_annot(rect)`
2. Execute `page.apply_redactions(images=pymupdf.PDF_REDACT_IMAGE_NONE, graphics=pymupdf.PDF_REDACT_LINE_ART_REMOVE_IF_TOUCHED, pymupdf.PDF_REDACT_TEXT_NONE)`.

This should remove all vector graphics inside / intersecting all `rect`s and will not touch any images or text.
